src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -599,48 +632,98 @@ protected void deleteNetworks(Azure azureService, List<com.microsoft.azure.manag
+            //.addListener(Runnables.doNothing(), Executors.newSingleThreadExecutor());
+            try {
+                vmfuture.get(TIMEOUT_SCRIPT_EXECUTION, TimeUnit.MINUTES);
+            } catch (InterruptedException e) {
+                log.info("An interruption occurred when updating extension to execute script on a VM.");
+                e.printStackTrace();
+            } catch (ExecutionException e) {
+                log.info("An error occurred when updating extension to execute script on a VM.");
+            } catch (TimeoutException e) {
+                log.info("Timeout reached when updating extension to execute script on a VM.");
+            }
             log.debug("Execution of script has been requested.");
+
         } else {
             log.info("Request Azure provider to install script extension and to execute script: " +
-                     concatenatedScripts.toString());
-            vm.update()
-              .defineNewExtension(createUniqueScriptName(vm.name()))
-              .withPublisher(SCRIPT_EXTENSION_PUBLISHER)
-              .withType(SCRIPT_EXTENSION_TYPE)
-              .withVersion(SCRIPT_EXTENSION_VERSION)
-              .withMinorVersionAutoUpgrade()
-              .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY, concatenatedScripts.toString())
-              .attach()
-              .apply();
-            log.debug("Installation of script extension and execution of script has been requested.");
+                        concatenatedScripts.toString());
+            if (vm.osType().equals(OperatingSystemTypes.LINUX)) {
+                AbstractFuture<VirtualMachine> vmfuture = vm.update()
+                                                            .defineNewExtension(createUniqueScriptName(vm.name()))
+                                                            .withPublisher(SCRIPT_EXTENSION_PUBLISHER_LINUX)
+                                                            .withType(SCRIPT_EXTENSION_TYPE_LINUX)
+                                                            .withVersion(SCRIPT_EXTENSION_VERSION_LINUX)
+                                                            .withMinorVersionAutoUpgrade()
+                                                            .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY,
+                                                                               concatenatedScripts.toString())
+                                                            .attach()
+                                                            .applyAsync(null);
+                try {
+                    vmfuture.get(TIMEOUT_SCRIPT_EXECUTION, TimeUnit.MINUTES);
+                } catch (InterruptedException e) {
+                    log.error("An interruption occurred while installing extension and executing script to a Linux VM.");
+                    Thread.currentThread().interrupt();
+                    e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)